File: software/examples/javascript/ExampleThreshold.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change to your UID

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var uvl = new Tinkerforge.BrickletUVLight(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Get threshold callbacks with a debounce time of 10 seconds (10000ms)
        uvl.setDebouncePeriod(10000);

        // Configure threshold for UV light "greater than 750 µW/cm²" (unit is µW/cm²)
        uvl.setUVLightCallbackThreshold('>', 750, 0);
    }
);

// Register UV light reached callback
uvl.on(Tinkerforge.BrickletUVLight.CALLBACK_UV_LIGHT_REACHED,
    // Callback function for UV light reached callback (parameter has unit µW/cm²)
    function (uvLight) {
        console.log('UV Light: ' + uvLight + ' µW/cm²');
        console.log('UV Index > 3. Use sunscreen!');
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/csharp/ExampleThreshold.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your UV Light Bricklet

	// Callback function for UV light reached callback (parameter has unit µW/cm²)
	static void UVLightReachedCB(BrickletUVLight sender, long uvLight)
	{
		Console.WriteLine("UV Light: " + uvLight + " µW/cm²");
		Console.WriteLine("UV Index > 3. Use sunscreen!");
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletUVLight uvl = new BrickletUVLight(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get threshold callbacks with a debounce time of 10 seconds (10000ms)
		uvl.SetDebouncePeriod(10000);

		// Register UV light reached callback to function UVLightReachedCB
		uvl.UVLightReachedCallback += UVLightReachedCB;

		// Configure threshold for UV light "greater than 750 µW/cm²" (unit is µW/cm²)
		uvl.SetUVLightCallbackThreshold('>', 750, 0);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}
